dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -866,7 +865,15 @@ Spdp::handle_auth_request(const DDS::Security::ParticipantStatelessMessage& msg)
     // We're simply caching this for later, since we can't actually do much without the SPDP announcement itself
     pending_remote_auth_tokens_[guid] = msg.message_data[0];
   } else {
-    iter->second.remote_auth_request_token_ = msg.message_data[0];
+    DiscoveredParticipant& dp = iter->second;
+    if (!(dp.remote_auth_request_token_ == msg.message_data[0])) {
+      dp.remote_auth_request_token_ = msg.message_data[0];
+      if (dp.is_requester_) {
+        dp.handshake_state_= DCPS::HANDSHAKE_STATE_EXPECTING_REQUEST;
+      } else {
+        send_handshake_request(guid, dp);
```

Comment:
Doesn't this need to check the auth state / handshake state before sending off a handshake request? Currently, it seems like this path can circumvent calling validate_remote_identity before sending the handshake request, which seems contrary to the state machine in Figure 9 of the security spec.

Reply:
Correct.  Fixed.